documentation/asciidoc/topics/yaml/grafana_datasource.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDataSource
+metadata:
+  name: grafanadatasource
+spec:
+  name: datasource.yaml
+  datasources:
+    - access: proxy
+      editable: true
+      isDefault: true
+      jsonData:
+        httpHeaderName1: Authorization
+        timeInterval: 5s
+        tlsSkipVerify: true
+      name: Prometheus
+      secureJsonData:
+        httpHeaderValue1: >-
+          Bearer
+          eyJhbGciOiJSUzI1NiIsImtpZCI6Imc4O...
+      type: prometheus
+      url: >-
+        https://thanos-querier-openshift-monitoring.apps.rhdg.prod.net
```

Comment:
Why do you need to go to the external route from the cluster, when it then goes back inside?
Should be `url: 'https://thanos-querier.openshift-monitoring.svc.cluster.local:9091'`

Also we need automate with the operator code or create as set number of templates for users. I've done all these steps, but they are bit difficult for regular users, large number of copy/paste complex values.

https://www.openshift.com/blog/thanos-querier-versus-thanos-querier